File: Project/javaScript/src/TaskStatus.js
/**
 * Task status constants for lifecycle tracking.
 * @readonly
 */
export const PENDING = 'PENDING';
export const IN_PROGRESS = 'IN_PROGRESS';
export const COMPLETED = 'COMPLETED';

/** @type {readonly string[]} */
export const ALL_STATUSES = Object.freeze([PENDING, IN_PROGRESS, COMPLETED]);

/**
 * @param {string} value
 * @returns {boolean}
 */
export function isValidStatus(value) {
  return typeof value === 'string' && ALL_STATUSES.includes(value);
}

export default {
  PENDING,
  IN_PROGRESS,
  COMPLETED,
  ALL_STATUSES,
  isValidStatus,
};

File: Project/java/src/main/java/com/basanta/todo/TaskStatus.java
package com.basanta.todo;

import java.util.List;

/**
 * Task status constants for lifecycle tracking.
 */
public final class TaskStatus {
    public static final String PENDING = "PENDING";
    public static final String IN_PROGRESS = "IN_PROGRESS";
    public static final String COMPLETED = "COMPLETED";

    public static final List<String> ALL_STATUSES = List.of(PENDING, IN_PROGRESS, COMPLETED);

    private TaskStatus() {}

    public static boolean isValidStatus(String value) {
        return value != null && ALL_STATUSES.contains(value);
    }
}
